### ISSUE TITLE
Failing unit tests: mysqli::real_escape_string() expects parameter 1 to be string, array given

### ISSUE THREAD

**Issue by owenherbert** [Wunderbyte-GmbH/moodle-mod_booking#333]
**Describe the bug**
The plugin unit tests are failing.

**Versions uses**
Moodle version: Moodle 4.1
Boooking version: main (687076c4556ce4893c71aa9edf6b37744c3d82fe)

**To Reproduce**
Run plugin unit tests.

**Expected behavior**
The unit tests should pass.

**Screenshots**
```
1) core\externallib_test::test_all_external_info with data set "mod_booking_bookings" (stdClass Object (...))
moodle_exception: Unsupported redirect detected, script execution terminated

/var/www/site/lib/weblib.php:2933
/var/www/site/lib/moodlelib.php:2787
/var/www/site/mod/booking/externallib.php:34
/var/www/site/lib/externallib.php:91
/var/www/site/lib/tests/externallib_test.php:442
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97

2) mod_booking\booking_option_test::test_delete_responses_activitycompletion
mysqli::real_escape_string() expects parameter 1 to be string, array given

/var/www/site/lib/dml/mysqli_native_moodle_database.php:1154
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1366
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1417
/var/www/site/mod/booking/lib.php:449
/var/www/site/course/modlib.php:137
/var/www/site/lib/testing/generator/module_generator.php:278
/var/www/site/mod/booking/tests/generator/lib.php:72
/var/www/site/lib/testing/generator/data_generator.php:503
/var/www/site/mod/booking/tests/booking_option_test.php:76
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97

3) mod_booking\events_test::test_teacher_added
mysqli::real_escape_string() expects parameter 1 to be string, array given

/var/www/site/lib/dml/mysqli_native_moodle_database.php:1154
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1366
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1417
/var/www/site/mod/booking/lib.php:449
/var/www/site/course/modlib.php:137
/var/www/site/lib/testing/generator/module_generator.php:278
/var/www/site/mod/booking/tests/generator/lib.php:72
/var/www/site/lib/testing/generator/data_generator.php:503
/var/www/site/mod/booking/tests/events_test.php:69
/var/www/site/mod/booking/tests/events_test.php:99
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97

4) mod_booking\events_test::test_teacher_removed
mysqli::real_escape_string() expects parameter 1 to be string, array given

/var/www/site/lib/dml/mysqli_native_moodle_database.php:1154
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1366
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1417
/var/www/site/mod/booking/lib.php:449
/var/www/site/course/modlib.php:137
/var/www/site/lib/testing/generator/module_generator.php:278
/var/www/site/mod/booking/tests/generator/lib.php:72
/var/www/site/lib/testing/generator/data_generator.php:503
/var/www/site/mod/booking/tests/events_test.php:69
/var/www/site/mod/booking/tests/events_test.php:130
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97

5) mod_booking\lib_test::test_subscribe_teacher_to_booking_option
mysqli::real_escape_string() expects parameter 1 to be string, array given

/var/www/site/lib/dml/mysqli_native_moodle_database.php:1154
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1366
/var/www/site/lib/dml/mysqli_native_moodle_database.php:1417
/var/www/site/mod/booking/lib.php:449
/var/www/site/course/modlib.php:137
/var/www/site/lib/testing/generator/module_generator.php:278
/var/www/site/mod/booking/tests/generator/lib.php:72
/var/www/site/lib/testing/generator/data_generator.php:503
/var/www/site/mod/booking/tests/lib_test.php:66
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80
phpvfscomposer:///var/www/site/vendor/phpunit/phpunit/phpunit:97
```

**Additional context**
Environment information:
```
Moodle 4.1.6+ (Build: 20231116)
Php: 7.4.33, mariadb: 10.5.23, OS: Linux 6.2.0-1009-aws x86_64
PHPUnit 9.5.28 by Sebastian Bergmann and contributors.
```

**Comment by bernhard-wunderbyte** [comment]
These are bugs in Booking v7.8.7.02 which is a very old version. We are currently trying to publish Wunderbyte Table on moodle.org. Once this is done, we can then publish the much newer Booking 8 on moodle.org. We won't fix any Bugs in Booking 7 anymore.

If you want to try Booking 8, you can get it here (you'll need to install Wunderbyte Table first):
- https://github.com/Wunderbyte-GmbH/moodle-local_wunderbyte_table/releases/tag/v1.8.3
- https://github.com/Wunderbyte-GmbH/moodle-mod_booking/releases/tag/v8.0.52
